src/VersionFeed.php

```diff
@@ -114,7 +115,16 @@ public function getDiffList($highestVersion = null, $limit = 100)
                 }
 
                 // Copy the link so it can be cached.
-                $version->GeneratedLink = $version->AbsoluteLink();
+                $oldPage = $version->getField('object');
+                if (!$oldPage instanceof SiteTree) {
```

Comment:
Feels like Versioned_Version should fall back to `new $baseclass` not `new DataObject`.